Tests/Common/Util/RateGateTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Tests.Common.Util
+{
+    [TestFixture]
+    public class RateGateTests
+    {
+        [Test, Ignore("Running in combination with multiple tests causes this to fail")]
```

Comment:
I don't think I can get this one to pass, even if I'm generous with the amount of time I give it. I just ran a test in conjunction with other tests and it resulted in it failing because it was 26 seconds late 😬

Reply:
> because it was 26 seconds late

That's a lot 😄, seems like there is another underlying issue, maybe a race condition? - The `RateGate` has 10 occurrences but the for loop goes from 0 to 10 inclusive, is this on purpose?

Reply:
Seems that `RateGate_200InstancesWaitOnAveragePlus150msMinus20ms` is still failing in the last commit